packages/enzyme/src/Utils.js

```diff
@@ -42,7 +42,7 @@ export function makeOptions(options) {
 }
 
 export function isCustomComponentElement(inst, adapter) {
-  return !!inst && adapter.isValidElement(inst) && typeof inst.type === 'function';
+  return adapter.isCustomComponentElement(inst);
```

Comment:
this needs a fallback for the case where an adapter lacks `isCustomComponentElement`.

Reply:
done,